src/ws-server.ts

```diff
@@ -0,0 +1,53 @@
+import axios from "axios";
+import { Pool } from "pg";
+import * as websockets from "ws";
+
+import { assertNever, contentTypeHeaders, graphqlEndpoint, UtilEither} from "./utils";
+
+const MSG_TYPE_RESTORE = "RESTORE";
+const ADDR_LENGTH_LIMIT = 2000;
+const ADDR_VALID_STARTS = ["Ddz", "Ae2"];
+
+/*
+  I am not sure this is equiv to 
+  SELECT DISTINCT utxos.receiver FROM utxos
+  on the tangata schema, just my best guess.
+*/
+const GRAPHQL_QUERY = `
+{
+  utxos(distinct_on: address) {
```

Comment:
Strictly speaking, we need all UTXOs where the address is a Daedalus address.

What is a Daedalus address you ask?

Cardano (starting in Shelley) has multiple address types:
- Base
- Enterprise
- Pointer
- Bootstrap

Bootstrap refers to the type of addresses used in the Byron era. In Yoroi, these addresses start with Ae2 and in Daedalus, these addresses start with Dd

Although this query will not fail if we return more addresses than needed, it's just less efficient (so don't spend too much time on this if it doesn't look easy to do). So preferably, we would only return bootstrap addresses that are of the Daedalus variety.
You can tell something is a bootstrap address if it starts with the bits `0b1000` (hopefully GraphQL provides at least this much). I don't suggest doing something like "starting starts with Dd" because the Dd prefix is network-specific (by design so that addresses look different on the testnet)

Sorry this is maybe a complex explanation. Let me know if you have any questions.

Reply:
you're right that this query is _slow_.  though it was the same process (filtering out addresses after you got it) as done in the old backend.  

I am confident I can do a graphql filter for strings starting with `Ae2` or `Dd` but I don't think I can check if it starts with a certain sequence of bytes.  the underlying sql tables and views hold addresses as `varchar`.  there's probably a useful postgresql function for the filter, but its not in graphql.

Reply:
this doesn't actually work.  the version of cardano-graphql we are using in production does not support `distinct_on: address` in a `utxos` query.  a more recent version does support it, but it breaks other things.  since we should be moving to the top of upstream anyway, I'll fix it on upstream and then come back to this.

Reply:
alas this does not work. Even balas with the PR to cardano-graphql.

If I run the following sql query on the schema (views) created by cardano-graphql I get lots of valid addresses:
`select distinct address from  "Utxo" where address like 'Ae2%';`

yet that same graphql query returns a completely different, smaller resultset.  notably, it lacks anything that looks like a bootstrap address.  everything in the graphql resultset starts with `2w1sdSJu3GVd`

Reply:
ah, I see the problem.  the query works fine if you do it on hasura directly, but the cardano-graphql node app actually truncates the query.  hmm.  tough nuggets.  because the result set is so big it might be best to do this in sql directly afterall.